persist/fs/types.go

```diff
@@ -70,6 +70,24 @@ type FileSetReader interface {
 	EntriesRead() int
 }
 
+// FileSetSeeker provides an out of order reader for a TSDB file set
+type FileSetSeeker interface {
+	io.Closer
+
+	// Open opens the files for the given shard and version for reading
+	Open(namespace ts.ID, shard uint32, start time.Time) error
+
+	// Seek returns the data for specified id provided the index was loaded upon open. An
+	// error will be returned if the index was not loaded or id cannot be found
+	Seek(id string) (data []byte, err error)
```

Comment:
We should make the `id` a `ts.ID` instead of `string` to be consistent with all our other interfaces.